docker/docker.go

```diff
@@ -279,21 +426,32 @@ func parseImageResponse(resp *http.Response, image *Image) error {
 		for i := range imageV2.Layers {
 			image.FsLayers[i].BlobSum = imageV2.Layers[i].Digest
 		}
-		image.digest = imageV2.Config.Digest
+		image.Digest = imageV2.Config.Digest
 		image.schemaVersion = imageV2.SchemaVersion
 	case "application/vnd.docker.distribution.manifest.v1+prettyjws":
-		var imageV1 imageV1
-		if err := json.NewDecoder(resp.Body).Decode(&imageV1); err != nil {
-			fmt.Fprintln(os.Stderr, "ImageV1 decode error")
-			return err
+		body, err := ioutil.ReadAll(resp.Body)
+		schema1, err := docker_manifest.Schema1FromManifest(body)
+		if err != nil {
+			return fmt.Errorf("failed to convert schema1 from manifest: %v", err)
 		}
-		image.FsLayers = make([]FsLayer, len(imageV1.FsLayers))
-		// in schemaVersion 1 layers are in reverse order, so we save them in the same order as v2
+		for i, compatibility := range schema1.ExtractedV1Compatibility {
+			log.Debugf("layer %v: cmd: %v", i, compatibility.ContainerConfig.Cmd)
+		}
+		if len(schema1.FSLayers) != len(schema1.ExtractedV1Compatibility) {
+			return fmt.Errorf("number of layers(%v) doesn't match the number of commands(%v)", len(schema1.FSLayers), len(schema1.ExtractedV1Compatibility))
+		}
+		image.FsLayers = make([]FsLayer, len(schema1.FSLayers))
+		image.FsCommands = make([]*FsLayerCommand, len(schema1.FSLayers))
+		// in schemaVersion 1 layers and commands are in reverse order, so we save them in the same order as v2
 		// base layer is the first
-		for i := range imageV1.FsLayers {
-			image.FsLayers[len(imageV1.FsLayers)-1-i].BlobSum = imageV1.FsLayers[i].BlobSum
+		for i := range schema1.FSLayers {
+			image.FsLayers[len(schema1.FSLayers)-1-i].BlobSum = schema1.FSLayers[i].BlobSum.String()
+			image.FsCommands[len(schema1.FSLayers)-1-i] = &FsLayerCommand{
+				Command: strings.Join(schema1.ExtractedV1Compatibility[len(schema1.FSLayers)-1-i].ContainerConfig.Cmd,","),
+				Layer:   schema1.FSLayers[i].BlobSum.Hex(),
+			}
```

Comment:
extract to func and unit test, it's not so trivial I think.

Reply:
Thanks, this caught a bug:
Should be: strings.Join(schema1.ExtractedV1Compatibility[i]